### PR TITLE
Converter issues with C2A

### PR DESCRIPTION
**Pull request by sharanagoudapatil**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #<gh-issue-id>

Test URLs:
- Before: https://main--danaher-ls-aem--hlxsites.hlx.page/
- After: https://452-converter-issues-with-c2a--danaher-ls-aem--hlxsites.hlx.page/

### CODE REVIEW

File: tools/actions/convert/test/fixtures/topic-hub-semantic.html [hunk 1]
```diff
@@ -31,6 +31,7 @@ <h1>Assay Kits</h1>
         <div>
           <div>
             <h2>Explore our line of Assay Kits</h2>
+            <p>View Products</p>
```
Comment (mhaack) [inline tools/actions/convert/test/fixtures/topic-hub-semantic.html:34]: I think this is not correct? @duynguyen you worked on the topic hub and this is already converted into a block right?

File: tools/actions/convert/test/fixtures/blog.html [hunk 2]
```diff
@@ -36,26 +38,197 @@
...
-<meta property="og:site_name" content="Danaher Life Sciences"/>
-<meta property="og:url" content="https://stage.lifesciences.danaher.com/content/danaher/ls/us/en/blog/key-steps-to-unlock-mrna-capability-beyond-covid.html"/>
 
 <script>
-dataLayer = [{
-    'user': {"customerID":"","accountType":"guest","marketCode":"","company":"","role":"","city":"","state":"","country":"","postalCode":"","lastVisit":""}
-},{
-    'page': {"title":"Unlocking mRNA's Potential Beyond COVID-19 | Danaher Life Sciences","language":"en","locale":"US","level":"top","type":"webpage","keywords":"","creationDate":"Dec 16, 2022 10:55:47 AM","updateDate":"Nov 06, 2023 11:56:39 AM"}
-}];
+    DanaherConfig = {
```
Comment (mhaack) [inline tools/actions/convert/test/fixtures/blog.html:47]: We should not have this here, no DanaherConfig, Target Key, GTM Ids etc. in test content

File: tools/importer/transformers/callToAction.js [hunk 3]
```diff
@@ -5,17 +5,33 @@ const createCTA = (main, document) => {
...
+    const btn2 = document.createElement('button');
+    btn2.textContent = btnText2;
+    if (rfqBtn2 && btn2.textContent) {
+      div.append(btn2);
+    }
+    if (btnhref2 && btn2.textContent) {
+      div.append(btn2);
+    }
+
     const cells = [
```
Comment (mhaack) [inline tools/importer/transformers/callToAction.js:35]: I cannot finde a test case for this. We should have tests with C2A with one and C2A with two buttons.

File: tools/importer/transformers/callToAction.js [hunk 4]
```diff
@@ -5,17 +5,33 @@ const createCTA = (main, document) => {
...
+    }
+    const btn2 = document.createElement('button');
+    btn2.textContent = btnText2;
+    if (rfqBtn2 && btn2.textContent) {
+      div.append(btn2);
+    }
+    if (btnhref2 && btn2.textContent) {
+      div.append(btn2);
+    }
+
```
Comment (mhaack) [inline tools/importer/transformers/callToAction.js:34]: This seams append `bt1` & `bt2` twice? What is the difference here?

File: tools/actions/convert/test/fixtures/topic-hub-semantic.html (hunk 1 above)
Comment (duynguyen) [reply tools/actions/convert/test/fixtures/topic-hub-semantic.html:34]: in the topics hub I didn't touch the CTA. I did see that the CTA converter was broken / not giving correct HTML if the button link is a normal URL, not a quote.
so having "view products" now is good, but shouldn't there be a URL as well?

Comment (duynguyen) [reply tools/actions/convert/test/fixtures/topic-hub-semantic.html:34]: we might need to update the cta block on the frontend too. For now it assumes the cta button opens a request for quote.

My proposal:

- `<a href="#request-quote">Request For Quote</a>` should trigger the quote modal.
- `<a href="link/to/products/hub">View Products</a>` should just go to the URL in the <a> element

File: tools/importer/transformers/callToAction.js (hunk 4 above)
Comment (sharanagoudapatil) [reply tools/importer/transformers/callToAction.js]: btn1 and btn2 - both can have RFQ Model link or page URL/Content paths

File: tools/actions/convert/test/fixtures/blog.html (hunk 2 above)
Comment (sharanagoudapatil) [reply tools/actions/convert/test/fixtures/blog.html]: DanaherConfig and GTM scripts were removed now. Please verify.

File: tools/actions/convert/test/fixtures/topic-hub-semantic.html (hunk 1 above)
Comment (sharanagoudapatil) [reply tools/actions/convert/test/fixtures/topic-hub-semantic.html]: It is updated in the converter and updated the tests. Please verify.

File: tools/importer/transformers/callToAction.js (hunk 4 above)
Comment (mhaack) [reply tools/importer/transformers/callToAction.js]: Ok understood

File: tools/importer/transformers/workflowContainer.js [hunk 5]
```diff
@@ -11,7 +11,7 @@ const createWorkflowContainerSection = (main, document) => {
     if (i === arr.length - 1) {
       const cells = [['Section Metadata'], ['style', 'container-two-col']];
       const table = WebImporter.DOMUtils.createTable(cells, document);
-      e.append(table, document.createElement('hr'));
+      e.append(table);
```
[the remark this answers is not in the document]
Comment (rgravitvl) [reply tools/importer/transformers/workflowContainer.js:14]: Yes @mhaack , but this place all the time before CTA Section

File: tools/importer/transformers/callToAction.js (hunk 3 above)
Comment (sharanagoudapatil) [reply tools/importer/transformers/callToAction.js]: @mhaack - I've updated the test cases for two buttons. Please verify.

File: tools/actions/convert/test/fixtures/topic-hub-semantic.html (hunk 1 above)
Comment (sharanagoudapatil) [reply tools/actions/convert/test/fixtures/topic-hub-semantic.html]: @mhaack & @duynguyen - Please review the changes. Thanks.

File: tools/actions/convert/test/fixtures/blog-semantic.html [hunk 6]
```diff
@@ -17,8 +18,16 @@
       <h1>The AI-powered pixel classifier</h1>
       <p>Fast and reproducible microscopy-image segmentation results</p>
       <p>
-        <img src="https://author-p93411-e849602.adobeaemcloud.com/content/dam/danaher/blogs/internal-blog/csm-pixel-classifier-video-nspshot.jpg" alt="">
+        <img src="https://author-p93411-e849602.adobeaemcloud.com/content/dam/danaher/blogs/internal-blog/csm-pixel-classifier-video-nspshot.jpg" alt="null">
```
Comment (rgravitvl) [inline tools/actions/convert/test/fixtures/blog-semantic.html:21]: It should not null, rather it can be empty.

**Comment by rgravitvl** [comment]
I have one minor comment, rest looks good to me.

**Review by rgravitvl — APPROVED** [review]

File: tools/actions/convert/test/fixtures/blog-semantic.html (hunk 6 above)
Comment (sharanagoudapatil) [reply tools/actions/convert/test/fixtures/blog-semantic.html]: I've updated the tests. Thanks.

File: tools/importer/transformers/util.js [hunk 7]
```diff
@@ -161,3 +161,54 @@ export const testimonial = (testimonialElement, document) => {
   const table = WebImporter.DOMUtils.createTable(block, document);
   testimonialEl.replaceWith(table);
 };
+
+export const c2a = (cta, document) => {
```
Comment (duynguyen) [inline tools/importer/transformers/util.js:165]: wondering if the `cta` rendering should be moved to util, which is intended for shared functions.
If it's only used by the `callToAction` block, we should keep it in the same script.

Comment (sharanagoudapatil) [reply tools/importer/transformers/util.js]: Moved c2a function from util.js to callToAction.js.

**Review by mhaack — APPROVED** [review]

**Review by duynguyen — APPROVED** [review]

**Review by davenichols-DHLS — APPROVED** [review]
Looks like it's good and happy to merge.

**Pull request merged**